pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -0,0 +1,501 @@
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/clock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/ratelimiter"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// finalizerName is the backupbucket controller finalizer.
+const finalizerName = "core.gardener.cloud/backupbucket"
+
+// RequeueDurationWhenResourceDeletionStillPresent is the duration used for requeueing when owned resources are still in
+// the process of being deleted when deleting a BackupBucket.
+var RequeueDurationWhenResourceDeletionStillPresent = 5 * time.Second
+
+// Reconciler reconciles the BackupBuckets.
+type Reconciler struct {
+	GardenClient    client.Client
+	SeedClient      client.Client
+	Config          config.BackupBucketControllerConfiguration
+	Clock           clock.Clock
+	Recorder        record.EventRecorder
+	GardenNamespace string
+	SeedName        string
+
+	// RateLimiter allows limiting exponential backoff for testing purposes
+	RateLimiter ratelimiter.RateLimiter
+}
+
+// Reconcile reconciles the BackupBuckets and deploys extensions.gardener.cloud/v1alpha1.BackupBucket in the seed cluster.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	backupBucket := &gardencorev1beta1.BackupBucket{}
+	if err := r.GardenClient.Get(ctx, request.NamespacedName, backupBucket); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Object is gone, stop reconciling")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
+	}
+
+	extensionBackupBucket := &extensionsv1alpha1.BackupBucket{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: backupBucket.Name,
+		},
+	}
+
+	if v1beta1helper.HasOperationAnnotation(backupBucket.Annotations) {
+		if updateErr := r.removeGardenerOperationAnnotation(ctx, backupBucket); updateErr != nil {
+			return reconcile.Result{}, fmt.Errorf("could not remove %q annotation: %w", v1beta1constants.GardenerOperation, updateErr)
+		}
+	}
```

Comment:
Hm, why do we have to do this? Shouldn't we delete the operation annotation in the API server like for other resources (`Shoot`, `ManagedSeed`) instead of doing it in the controller (ref  https://github.com/gardener/gardener/blob/f6a4f13ecc3c10460d87d9c7826b2f5fd8b35cd4/pkg/registry/seedmanagement/managedseed/strategy.go#L88)? 
Otherwise, if the controller crashes after this before updating the `status` it wouldn't reconcile the object when coming up again.